--- js.mak	2007-10-19 14:57:52.000000000 -0400
+++ js.vs2008.mak	2008-10-04 16:38:39.941958500 -0400
@@ -105,13 +105,15 @@
 
 "$(OUTDIR)" :
     if not exist "$(OUTDIR)/$(NULL)" mkdir "$(OUTDIR)"
-
+NSPR=../../nspr
 CPP=cl.exe
 # ADD BASE CPP /nologo /MT /W3 /GX /O2 /D "WIN32" /D "NDEBUG" /D _X86_=1 /D "_WINDOWS" /YX /c
 # ADD CPP /nologo /MD /W3 /GX /O2 /D "NDEBUG" /D _X86_=1 /D "_WINDOWS" /D "WIN32" /D "XP_WIN" /D "JSFILE" /D "EXPORT_JS_API" /I"$(INTDIR)" /YX /c
-CPP_PROJ=/nologo /MD /W3 /GX /O2 /D "NDEBUG" /D _X86_=1 /D "_WINDOWS" /D "WIN32" /D\
- "XP_WIN" /D "JSFILE" /D "EXPORT_JS_API" /Fp"$(INTDIR)/js.pch" /I"$(INTDIR)" /YX\
- /Fo"$(INTDIR)/" /c 
+CPP_PROJ=/nologo /MD /W3 /EHca /O2 /D "NDEBUG" /D _X86_=1 /D "_WINDOWS" /D "WIN32" /D\
+ "XP_WIN" /D "JSFILE" /D "EXPORT_JS_API" /Fp"$(INTDIR)/js.pch" /I"$(INTDIR)" \
+ /Fo"$(INTDIR)/" /c /D "_CRT_SECURE_NO_DEPRECATE" \
+ /D "_CRT_NONSTDC_NO_DEPRECATE" /D "JS_THREADSAFE" \
+ /I $(NSPR)/include
 CPP_OBJS=.\Release/
 CPP_SBRS=.\.
 
@@ -190,11 +192,13 @@
 	"$(INTDIR)\jsxdrapi.obj" \
 	"$(INTDIR)\jsxml.obj" \
 	"$(INTDIR)\prmjtime.obj" \
+	"$(INTDIR)\jsiter.obj" \
 	"$(OUTDIR)\fdlibm.lib"
 
+LIBS=$(NSPR)/lib/libnspr4.lib
 "$(OUTDIR)\js32.dll" : "$(OUTDIR)" $(DEF_FILE) $(LINK32_OBJS)
     $(LINK32) @<<
-  $(LINK32_FLAGS) $(LINK32_OBJS)
+  $(LINK32_FLAGS) $(LINK32_OBJS) $(LIBS)
 <<
 
 !ELSEIF  "$(CFG)" == "js - Win32 Debug"
